Receiver/Receiver.cpp

```diff
@@ -0,0 +1,281 @@
+		visitCount[hours][date - 1] = visitCount[hours][date - 1] + 1; 
+	}
+	
+}
+
+void split(vector<string>& result, string s1, char sep) {
+	string val = "";
+	for (unsigned int i = 0; i < s1.length(); i++)
+	{
+		string x(1, s1[i]);
+		if (s1[i] == sep) {
+			result.push_back(val);
+			val = "";
+		}
+		else {
+			val = val + x;
+		}
+	}
+	result.push_back(val);
+}
+
+void Receiver::getSenderData()
+{
+	cout << "Enter the format: ";
+	string senderString,senderFormat;
+	getline(cin, senderFormat);
+	cout <<"Footfall Data format used by Sender: "<< senderFormat<<endl;
+
+	while (getline(cin, senderString))
+	{
+		if (senderString == "")
+			break;
+
+		storeInBuffer(senderFormat, senderString);
+	}
+}
+
+void Receiver::storeInBuffer(string& senderFormat,string& senderString)
+{
+	map<string, int> recievedData;
+	
+	vector<string>senderFormatVector;
+	vector<string>senderStringVector;
+
+	split(senderFormatVector, senderFormat,' ');
+	split(senderStringVector, senderString, ' ');
+
+	for (unsigned int i = 0; i < senderFormatVector.size(); i++)
+	{
+		recievedData[senderFormatVector[i]] = stoi(senderStringVector[i]);
+	}
+
+	Data addToBuffer(recievedData["dd"], recievedData["mm"], recievedData["yyyy"], recievedData["hh"], recievedData["mn"]);
+	BufferFootfall.push_back(addToBuffer);
+
+	if (BufferFootfall.size() >=50)
+	{
+		storeCountInVector();
+		BufferFootfall.clear();
+	}
+}
+
+void Receiver::getAvgHourlyfootfall()
+{
+	for (int hour = 0; hour < 24; hour++)
+	{
+		float sum = 0;
+		int noofDays = visitCount[0].size();
+		for (int day = 0; day < noofDays; day++)
+		{
+			sum = sum + visitCount[hour][day];
+		}
+		hourlyAvgFootfall.push_back(sum/noofDays);
+		
+	}
+}
+
+void Receiver::getAvgHourlyfootfall(int noofHours,int noofDays)
+{
+	for (int hour = 0; hour < noofHours; hour++)
+	{
+		float sum = 0;
+		for (int day = 0; day < noofDays; day++)
+		{
+			sum = sum + visitCount[hour][day];
+		}
+		hourlyAvgFootfall.push_back(sum/noofDays);
+		cout <<"Avg. Count for Hour: "<<hour<<" is: "<< hourlyAvgFootfall[hour] <<endl;
+	}
+}
+
+void Receiver::getAvgDailyfootfall()
+{
+	for (unsigned int day= 0; day < 7; day++)
+	{
+		float sum,count;
+		sum = count = 0;
+		for (unsigned int i = day; i < visitCount[0].size(); i = i + 7)
+		{
+			sum = sum + visitCount[24][i];
+			count++;
+		}
+		dailyAvgFootfall.push_back(sum / count);
+		cout << "Avg. Count for Day " << day << " is: " << dailyAvgFootfall[day] << endl;
+	}
+	
+}
+
+int Receiver::getPeekDailyfootfallMontly() 
+{
+	int date = 1;
+	int max = visitCount[24][0];
+	for (unsigned int day = 0; day < visitCount[0].size(); day++)
+	{
+		if (visitCount[24][day] > max)
+		{
+			max = visitCount[24][day];
+			date = day + 1;
+		}
+	}
+	peakValue = max;
+	peakdate = date;
+	cout << "Date on which we will get maximum count: " << date << " maximum value " << max << endl;
+	return max;
+}
+
+void Receiver::StoreInCSV()
+{
+	fstream MyFile;
+	MyFile.open("HourlyAverage.csv", ios::out | ios::app);
```

Comment:
check if file exists before open? better to add the file IO in try, catch block to capture the exceptions.